### PR TITLE
Pholthaus/deprecated functions

### PR DESCRIPTION
**Pull request by pascalholthaus**
**What**: Added script_bugtraq_id() function check to CheckDeprecatedFunctions  plugin.

**Why**:

**How**:

**Checklist**:

- [x] Tests
- [ ] Conventional commit message
- [ ] Documentation

### CODE REVIEW

File: troubadix/plugins/deprecated_bugtraq_id.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# pylint: disable=fixme
```
Comment (y0urself) [inline troubadix/plugins/deprecated_bugtraq_id.py:18]:
```suggestion
```

File: tests/plugins/test_files/fail_bad_new_line.nasl [hunk 2]
```diff
@@ -49,4 +48,4 @@ if(description)
   script_tag(name:"solution_type", value:"VendorFix");
 
   exit(0);
-}aa
\ No newline at end of file
+}aa
```
Comment (y0urself) [inline tests/plugins/test_files/fail_bad_new_line.nasl:51]:
```suggestion
}

```

**Review by y0urself — CHANGES_REQUESTED** [review]

**Comment by y0urself** [comment]
Please add a proper PR Title and sqash message.
Eg. "Add: New Plugin `CheckDeprecatedBugtraqIDTag`"

**Comment by cfi-gb** [comment]
Is there a reason why this hasn't been added to the existing https://github.com/greenbone/troubadix/blob/main/troubadix/plugins/deprecated_functions.py plugin? Should be IMHO more then enough.

File: tests/plugins/test_files/fail_bad_new_line.nasl (hunk 2 above)
Comment (pascalholthaus) [reply tests/plugins/test_files/fail_bad_new_line.nasl:51]: need to be there

**Review by y0urself — APPROVED** [review]

File: troubadix/plugins/deprecated_bugtraq_id.py (hunk 1 above)
Comment (pascalholthaus) [reply troubadix/plugins/deprecated_bugtraq_id.py]: file removed

File: troubadix/plugins/deprecated_bugtraq_id.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+from typing import Iterator
+
+from troubadix.helper import SpecialScriptTag
+from troubadix.helper.patterns import get_special_script_tag_pattern
+from troubadix.plugin import FilePlugin, LinterError, LinterResult
+
+
+class CheckDeprecatedBugtraqId(FilePlugin):
+    name = "check_deprecated_dependency"
```
[the remark this answers is not in the document]
Comment (pascalholthaus) [reply troubadix/plugins/deprecated_bugtraq_id.py]: file removed

**Pull request merged**